### PR TITLE
Fix OCP rule routes_protected_by_tls

### PR DESCRIPTION
**Pull request by Vincent056**
The rules give false negative result on a route that does not have insecureEdgeTerminationPolicy, this PR uses a jq filter to filter out the null in insecureEdgeTerminationPolicy, so it gives the correct result.

### CODE REVIEW

File: applications/openshift/networking/routes_protected_by_tls/tests/all_route_with_one_null.pass.sh [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+yum install -y jq
+
+kube_apipath="/kubernetes-api-resources"
+
+mkdir -p "$kube_apipath/apis/route.openshift.io/v1"
+
+routes_apipath="/apis/route.openshift.io/v1/routes"
+
+# This file assumes that we have two routes outside the openshift/kube namespaces, and
```
Comment (rhmdnd) [inline applications/openshift/networking/routes_protected_by_tls/tests/all_route_with_one_null.pass.sh:13]: nit: this comment is unrelated to the tests in this patch, right?

File: applications/openshift/networking/routes_protected_by_tls/tests/all_route_with_one_null.pass.sh [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                "resourceVersion": "213342",
+                "uid": "26f97c95-7351-49b8-a0dd-f6cdd88283ae"
+            },
+            "spec": {
+                "host": "wordpress-test-no-rate-limit.apps.wenshen-51.devcluster.openshift.com",
+                "port": {
+                    "targetPort": "8080-tcp"
+                },
+                "tls": {
+                    "termination": "passthrough"
```
Comment (rhmdnd) [inline applications/openshift/networking/routes_protected_by_tls/tests/all_route_with_one_null.pass.sh:90]: Awesome - this should be testing the case we hit.

File: applications/openshift/networking/routes_protected_by_tls/rule.yml [hunk 3]
```diff
@@ -38,17 +40,16 @@ severity: medium
...
-    ocp_data: "true"
-    filepath: /apis/route.openshift.io/v1/routes?limit=500
-    yamlpath: ".items[:].spec.tls.insecureEdgeTerminationPolicy"
-    check_existence_yamlpath: ".items[:].spec.host"
-    entity_check: "all"
-    check_existence: "all_exist"
-    values:
-      - value: "^(None|Redirect)$"
-        operation: "pattern match"
+    name: yamlfile_value
```
Comment (rhmdnd) [inline applications/openshift/networking/routes_protected_by_tls/rule.yml:46]: nit: is this indented another layer?

File: applications/openshift/networking/routes_protected_by_tls/rule.yml [hunk 4]
```diff
@@ -38,17 +40,16 @@ severity: medium
 
 warnings:
 - general: |-
-    {{{ openshift_cluster_setting("/apis/route.openshift.io/v1/routes?limit=500") | indent(4) }}}
+    {{{ openshift_filtered_cluster_setting({'/apis/route.openshift.io/v1/routes': jqfilter}) | indent(4) }}}
```
Comment (rhmdnd) [inline applications/openshift/networking/routes_protected_by_tls/rule.yml:43]: Should we keep the query string for paging limit responses?

**Review by rhmdnd — COMMENTED** [review]
Looks good. Waiting on a cluster to test this out.

File: applications/openshift/networking/routes_protected_by_tls/rule.yml (hunk 4 above)
Comment (Vincent056) [reply applications/openshift/networking/routes_protected_by_tls/rule.yml:43]: I am not too sure about this one

File: applications/openshift/networking/routes_protected_by_tls/tests/all_route_with_one_null.pass.sh (hunk 1 above)
Comment (Vincent056) [reply applications/openshift/networking/routes_protected_by_tls/tests/all_route_with_one_null.pass.sh:13]: that's right, I am going to remove this one

**Comment by rhmdnd** [comment]
Tested this out with the PCI-DSS profile and it works. Thanks, Vincent!

**Comment by jan-cerny** [comment]
> Tested this out with the PCI-DSS profile and it works. Thanks, Vincent!

@rhmdnd What prevents us from merging this? :smile:

**Review by jhrozek — APPROVED** [review]
